### PR TITLE
Add event specific data for Event model

### PR DESCRIPTION
**Pull request by sbulat**
**WHAT?**
The Event model has the `event_specific_data` property. It could be very different depending on the `key` property for the event. This PR introduces all of the possible shapes for this event-specific data.

### CODE REVIEW

File: components/schemas/event_specific_data/Dunning-Step-Reached.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+title: Dunning Step Reached
+type: object
+properties:
+  dunner:
+    type: object
+    properties:
```
Comment (michalpierog) [inline components/schemas/event_specific_data/Dunning-Step-Reached.yaml:6]: You need to extract those nested properties, so something like:

```
title: Dunning Step Reached
type: object
properties:
  dunner:
    $ref: "./Dunner-Event-Data.yaml"
  current_step:
     $ref: "./Current-Step-Event-Data.yaml"
```
(or some similar names). The validation is not working fine for this and didn't detect those issues.

File: components/schemas/event_specific_data/Invoice-Line-Items-Data.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+type: array
```
Comment (michalpierog) [inline components/schemas/event_specific_data/Invoice-Line-Items-Data.yaml:1]: You do not need to extract array

```
title: Invoice Line Items-Data
type: object
properties:
uid:
      type: string
    title:
      type: string
    description:
      type: string
...
```
and then in Proforma-Invoice-Issued.yaml:

```
  line_items:
    type: array
    items:
        $ref: "./Invoice-Line-Items-Data.yaml"
```

Comment (michalpierog) [inline components/schemas/event_specific_data/Invoice-Line-Items-Data.yaml:1]: Is it different than existing Invoice-Line-Item?

Comment (sbulat) [reply components/schemas/event_specific_data/Invoice-Line-Items-Data.yaml:1]: Unfortunately, here we have some additional fields like `quantity_delta` or `pricing_details`, that's why I created new schema.

File: components/schemas/event_specific_data/Subscription-Group-Signup-Failure.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+title: Subscription Group Signup Failure
+type: object
+properties:
+  subscription_group:
+    $ref: "./Subscription-Group-Failure-Data.yaml"
+  customer:
+    type: "null"
```
Comment (michalpierog) [inline components/schemas/event_specific_data/Subscription-Group-Signup-Failure.yaml:7]: Missing type, if this is nullable add another list item with $ref or simple type.

Comment (sbulat) [reply components/schemas/event_specific_data/Subscription-Group-Signup-Failure.yaml:7]: The way this event is created `customer` here would be always `null`. Is it okay to use `type: "null"` here or should it be presented in a different way?

Comment (michalpierog) [reply components/schemas/event_specific_data/Subscription-Group-Signup-Failure.yaml:7]: Ok, so lets leave it as it is. If there would be some problem with generated code we would fix this when writing tests.

**Review by michalpierog — APPROVED** [review]
LGTM

**Pull request merged**